### PR TITLE
fix: menu click brings up canvas

### PR DESCRIPTION
**Pull request by jianwei1**
# Description

### Issue

clicking on the stepblocknode menu will bring the user to the canvas view

### CODE REVIEW

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/JourneyFlow.tsx [hunk 1]
```diff
@@ -358,6 +358,10 @@ export function JourneyFlow(): ReactElement {
     // else go through standard positioning logic below
 
     if (isClickOrTouch(event.timeStamp)) {
+      const target = event.target as HTMLElement
+      // if the clicked/tapped element is the node menu, don't call handlestepselection hook https://github.com/JesusFilm/core/pull/4736
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/JourneyFlow.tsx:362]:
```suggestion
      // if the clicked/tapped element is the node menu, don't call handleStepSelection hook https://github.com/JesusFilm/core/pull/4736
```

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNodeMenu/StepBlockNodeMenu.tsx [hunk 2]
```diff
@@ -65,7 +65,7 @@ export function StepBlockNodeMenu({
           }}
           data-testid="EditStepFab"
         >
-          <EllipsisIcon />
+          <EllipsisIcon id="StepBlockNodeMenu" />
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNodeMenu/StepBlockNodeMenu.tsx:68]: add a test that asserts this id is in this element.
add a comment above this line:
"// id should match target.id in <function name> function of apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/JourneyFlow.tsx "

Comment (Kneesal) [reply apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNodeMenu/StepBlockNodeMenu.tsx:68]: add a similar comment for the element that has "edit-step'"

**Review by Kneesal — CHANGES_REQUESTED** [review]

**Review by Kneesal — APPROVED** [review]

File: apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNodeMenu/StepBlockNodeMenu.spec.tsx [hunk 3]
```diff
@@ -78,4 +78,36 @@ describe('StepBlockNodeMenu', () => {
...
+    )
+
+    await waitFor(() => {
+      expect(screen.getByTestId('EditStepFab')).toBeInTheDocument()
+    })
+    const fab = screen.getByTestId('EditStepFab')
+    expect(fab).toHaveAttribute('id', 'edit-step')
+  })
+
+  it('should have StepBlockNodeMenu id on fab icon', async () => {
```
Comment (Kneesal) [inline apps/journeys-admin/src/components/Editor/Slider/JourneyFlow/nodes/StepBlockNode/StepBlockNodeMenu/StepBlockNodeMenu.spec.tsx:98]:
```suggestion
  it('should have StepBlockNodeMenuIcon id on fab icon', async () => {
```

**Review by Kneesal — CHANGES_REQUESTED** [review]

**Review by Kneesal — DISMISSED** [review]

**Review by DanChae03 — APPROVED** [review]

**Pull request merged**